### PR TITLE
[enhancement](remote) support local cache GC at the granularity of cache files

### PR DESCRIPTION
**Pull request by luozenglin**
# Proposed changes

Issue Number: close #xxx

## Problem summary

The current local cache gc at the granularity of segment, under 'sub_file_cache' may cause some old subfiles can not be gc in time. to solve this problem this pr supports gc at the granularity of cache files.

The idea is to use a two-level lru queue, with the first level queue at the segment granularity and the second level queue at the sub-cache file granularity. Each time clean up the second level queue top file in the first level queue top element.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.9 seconds
 load time: 453 seconds
 storage size: 17123356347 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221208045718_clickbench_pr_59901.html

File: be/src/io/cache/file_cache_manager.cpp [hunk 1]
```diff
@@ -175,16 +193,21 @@ void FileCacheManager::gc_file_caches() {
...
         for (size_t i = 0; i < contexts.size(); ++i) {
-            std::list<FileCachePtr> gc_file_list;
-            contexts[i].get_gc_file_caches(gc_file_list);
-            for (auto item : gc_file_list) {
-                std::shared_lock<std::shared_mutex> rdlock(_cache_map_lock);
-                // for dummy file cache, check already used or not again
-                if (item->is_dummy_file_cache() &&
-                    _file_cache_map.find(item->cache_dir().native()) != _file_cache_map.end()) {
-                    continue;
+            auto context = contexts[i];
```
[the remark this answers is not in the document]
Comment (luozenglin) [reply be/src/io/cache/file_cache_manager.cpp]: replaced with auto&. Thanks.

File: be/src/io/cache/sub_file_cache.cpp [hunk 2]
```diff
@@ -199,23 +208,29 @@ Status SubFileCache::_get_need_cache_offsets(size_t offset, size_t req_size,
...
     std::vector<size_t> timeout_keys;
     {
         std::shared_lock<std::shared_mutex> rlock(_cache_map_lock);
         for (std::map<size_t, int64_t>::const_iterator iter = _last_match_times.cbegin();
              iter != _last_match_times.cend(); ++iter) {
             if (time(nullptr) - iter->second > _alive_time_sec) {
                 timeout_keys.emplace_back(iter->first);
+            } else {
+                auto [cache_file, done_file] = _cache_path(iter->first);
+                _gc_lru_queue.push({iter->first, iter->second});
```
[the remark this answers is not in the document]
Comment (luozenglin) [reply be/src/io/cache/sub_file_cache.cpp:222]: _gc_lru_queue is used to gc no timeout cache when the cache file exceeds the disk limit.

**Review by yangzhg — APPROVED** [review]
LGTM

**Pull request merged**